### PR TITLE
ContainerService 2019-06-01 API 

### PR DESCRIPTION
**Pull request by xizha162**
Included in this PR:
1. Agent pool get versions API to get a list of available versions to use when creating agent pool
2. Agent pool get upgrade profile API to get a list of available versions to upgrade for an agent pool
3. Added node taints property to agent pool. 
4. Added ScaleSetPriority and ScaleSetEvicitionPolicy to agent pool
5. Added EnableNodePublicIP property to agent pool

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by yangl900 — APPROVED** [review]
LGTM!

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json [hunk 1]
```diff
@@ -729,6 +729,100 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/upgradeProfiles/default": {
```
Comment (antmarti-microsoft) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json]: Is this API valid for anything other than "default"?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json [hunk 2]
```diff
@@ -1308,6 +1402,25 @@
...
+        "scaleSetEvictionPolicy": {
+          "$ref": "#/definitions/ScaleSetEvictionPolicy",
+          "description": "ScaleSetEvictionPolicy to be used to specify eviction policy for low priority virtual machine scale set. Default to Delete."
+        },
+        "nodeTaints": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Taints added to new nodes during node pool create and scale."
```
Comment (antmarti-microsoft) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json]: Worth describing the format of a 'taint' in the description?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json (hunk 1 above)
Comment (xizhamsft) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json]: Nope. It is always default for now. This is the same behavior as managedclusters/upgradeProfiles/default.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json (hunk 2 above)
Comment (xizhamsft) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/managedClusters.json]: Good idea. Added an example taints taking from official kubernetes documentation

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/location.json [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerService/locations/{location}/orchestrators": {
+      "get": {
+        "tags": [
+          "ContainerServices"
+        ],
+        "operationId": "ContainerServices_ListOrchestrators",
```
Comment (jhendrixMSFT) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/location.json]: Orchestrators_List

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/location.json [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+          },
+          "description": "List of orchestrator version profiles."
+        }
+      },
+      "required": [
+        "orchestrators"
+      ],
+      "description": "The properties of an orchestrator version profile."
+    },
+    "OrchestratorVersionProfileListResult": {
```
Comment (jhendrixMSFT) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-06-01/location.json]: The type name is a bit long, how about OrchestratorListResult?  Matches with the (new) operationId as well.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

**Review by jhendrixMSFT — APPROVED** [review]
Per discussion these names are copied from the previous API version so we will leave them as-is.

**Comment by jhendrixMSFT** [comment]
@antmarti-microsoft if your issues have been addresses can you please approve?  Your request for changes is preventing merging of this PR.

**Review by anthony-c-martin — APPROVED** [review]

**Pull request merged**